src/Plugin/rest/resource/AlertsRestResource.php

```diff
@@ -195,11 +208,34 @@ public function get() {
     }
 
     $uri = $this->request->query->get('uri');
+
+    // Extract the language code from the URI.
+    $langcode = substr($uri, 1, 2);
```

Comment:
@andrebonon is there a more bulletproof way to get the language? Do we know the url will always always look the same?

Reply:
Let's delve into the underlying issue here. The initial request constructs the entire page based on the detected language, which, in this case, is determined through the URL path. However, the subsequent request to /alert does not include the language in the path, leading Drupal to default to the primary language instead of maintaining the language from the original request.

My attempts to override the language for the /alert request were unsuccessful. 
Therefore, I opted to redirect the /alert request to /{lang}/alert, enabling Drupal to correctly identify the appropriate translation for the request.

It's important to note that this solution specifically impacts the Language detection URL when utilizing the path option. It doesn't affect the domain option, session, or any other existing detection modes.

Reply:
For the way to get the language, the other way I can think of would be using explode().
$path = explode('/', trim($uri, '/')); and get the $path[0].
That would help in cases we have custom lang codes or codes with 1 or 3 characters (what I guess don't follow ISO for lang codes).

Do you have any other suggestions?